src/components/execution/ExecutionTable.js

```diff
@@ -89,9 +100,35 @@ class ExecutionTable extends Component {
     const { executions, dispatchGetResults } = this.props;
     if (!prevExecutions.equals(executions)) {
       dispatchGetResults();
+
+      const mainPipelineExecutions = executions.filter(
+        (item) => !item.pipelineExecutionId,
+      );
+
+      const structuredExecutions = [...mainPipelineExecutions.toArray()];
+      structuredExecutions.forEach((v) => {
+        const element = v;
+        element.resultPipeline = executions
+          .filter((item) => item.pipelineExecutionId === v.id)
```

Comment:
`groupBy` could go well in your code here and/or a bit above. It exists in immutable as well https://immutable-js.github.io/immutable-js/docs/#/Map/groupBy